fdb-extensions/src/test/java/com/apple/foundationdb/async/RangeSetTest.java

```diff
@@ -233,66 +240,182 @@ public void insert() {
     }
 
     @Test
-    public void insertEmpty() {
+    void insertEmpty() {
         assertThrows(IllegalArgumentException.class, () -> rs.insertRange(db, new byte[0], new byte[]{0x00}));
     }
 
     @Test
-    public void insertInverted() {
+    void insertInverted() {
         assertThrows(IllegalArgumentException.class, () -> rs.insertRange(db, new byte[]{0x15}, new byte[]{0x05}));
     }
 
     @Test
-    public void insertPastEnd() {
+    void insertPastEnd() {
         assertThrows(IllegalArgumentException.class, () -> rs.insertRange(db, new byte[]{(byte)0xff, (byte)0x10}, new byte[]{(byte)0xff, (byte)0x15}));
     }
 
+    @Test
+    void readAndEditNonOverlappingRanges() {
+        try (MultipleTransactions multi = MultipleTransactions.create(db, 3)) {
+            Range r1 = new Range(new byte[]{0x00}, new byte[]{0x01});
+            Range r2 = new Range(new byte[]{0x02}, new byte[]{0x03});
+            Range r3 = new Range(new byte[]{0x04}, new byte[]{0x05});
+
+            // Insert r1
+            Transaction tr1 = multi.get(0);
+            assertTrue(rs.insertRange(tr1, r1, true).join());
+
+            // In separate transactions, read the ranges r2 and r3. One of them
+            // is before r1 and the other other after
```

Comment:
Aren't both after r1?

Reply:
Yeah, it seems like this is missing coverage of querying before the inserted range.